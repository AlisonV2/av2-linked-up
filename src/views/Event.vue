<template>
  <div class="container profile-container">
    <div class="row mb-3">
      <app-title>{{ event.name }}</app-title>
    </div>
    <div class="row">
      <div class="col-12 col-lg-5 img-center">
        <img class="img-fluid" :src="event.thumbnail" :alt="event.name" />
      </div>
      <div class="col-12 col-lg-7 text-center mb-2">
        <p class="accent-text">Description</p>
        <p>{{ event.description }}</p>
        <p>
          <span class="accent-text">Location :</span> {{ event.city }} ({{
            event.zip
          }})
        </p>
        <div class="col-12">
          <p><span class="accent-text">Date :</span> {{ event.date }}</p>
        </div>
        <div class="btn-group">
          <div
            class="mt-4 btn-center px-2 attend-event"
            @click="attendEvent"
            v-if="role === 'client' || role === 'artist'"
          >
            <app-button>Attend Event</app-button>
          </div>
          <div
            class="mt-4 btn-center book-event"
            @click="bookEvent"
            v-if="role === 'artist'"
          >
            <app-button>Book a stand</app-button>
          </div>
        </div>
      </div>
    </div>
    <div class="row mb-3">
      <app-title>Participating artists</app-title>
    </div>
    <div class="row artists-list">
      <div
        class="col-12 col-md-6 col-lg-3 col-xl-3"
        v-for="artist in artists"
        :key="artist.name"
      >
        <router-link :to="{ name: 'Profile', params: { id: artist.uid } }">
          <ArtistItem :artist="artist" />
        </router-link>
      </div>
    </div>
  </div>
</template>

<script>
import ArtistItem from '@/components/artists/ArtistItem';

/**
 * @exports EventPage
 * @type {Page}
 * @vue-data {object} event
 * @vue-data {boolean} attend
 * @vue-data {boolean} book
 * @vue-data {string} role
 * @vue-data {string} tickets
 * @vue-data {string} stands
 */
export default {
  name: 'Event',
  components: {
    ArtistItem,
  },
  data() {
    return {
      event: {
        id: '',
        name: '',
        description: '',
        date: '',
        zip: '',
        city: '',
        thumbnail: '',
        attendees: [],
        stands: [],
      },
      role: '',
      artists: [],
    };
  },
  created() {
    this.$store.dispatch('getEventById', this.$route.params.id).then(() => {
      this.event = this.$store.getters.getEventById;
    });

    this.$store.dispatch('getUserRole').then(() => {
      this.role = this.$store.getters.getUserRole;
    });

    this.$store.dispatch('getEventArtists', this.$route.params.id).then(() => {
      this.artists = this.$store.getters.getEventArtists;
      console.log(this.artists);
    });
  },
  methods: {
    attendEvent() {
      this.$router.push({
        name: 'Participation',
        params: { id: this.$route.params.id },
      });
    },
    bookEvent() {
      this.$router.push({
        name: 'Booking',
        params: { id: this.$route.params.id },
      });
    },
  },
};
</script>
