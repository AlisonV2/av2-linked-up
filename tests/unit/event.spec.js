import { shallowMount } from '@vue/test-utils';
import { createStore } from 'vuex';
import Event from '@/views/Event.vue';
import AppTitle from '@/components/app/Title';

const actions = {
  getEventById: jest.fn(),
};

const getters = {
  getEventById: jest.fn(),
};

const store = createStore({
  actions,
  getters,
});

const $route = {
  params: { id: 1 },
};

const config = {
  data() {
    return {
      event: {
        id: '',
        name: '',
        description: '',
        date: '',
        zip: '',
        city: '',
        thumbnail: '',
      },
    };
  },
  global: {
    plugins: [store],
    components: {
      'app-title': AppTitle,
    },
    mocks: {
      $route,
    },
  },
};

/**
 * @module EventTest
 */
describe('Event.vue', () => {
  it('Event Snapshot', () => {
    const component = shallowMount(Event, config);
    expect(component.element).toMatchSnapshot();
  });

  it('getEventById action should have been called', async () => {
    const component = shallowMount(Event, config);
    Event.created.call(component.vm);
    await component.vm.$nextTick();
    expect(actions.getEventById).toHaveBeenCalled;
  });

  it('getEventById getter should have been called', async () => {
    const component = shallowMount(Event, config);
    Event.created.call(component.vm);
    await component.vm.$nextTick();
    expect(getters.getEventById).toHaveBeenCalled;
  });
});
