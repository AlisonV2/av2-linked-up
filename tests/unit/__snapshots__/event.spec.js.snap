// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Event.vue Event Snapshot 1`] = `
<div
  class="container profile-container"
>
  <div
    class="row mb-3"
  >
    <app-title-stub />
  </div>
  <div
    class="row"
  >
    <div
      class="col-12 col-lg-5 img-center"
    >
      <img
        alt=""
        class="img-fluid"
        src=""
      />
    </div>
    <div
      class="col-12 col-lg-7 text-center mb-2"
    >
      <p
        class="accent-text"
      >
        Description
      </p>
      <p />
      <p>
        <span
          class="accent-text"
        >
          Location :
        </span>
          () 
      </p>
      <div
        class="col-12"
      >
        <p>
          <span
            class="accent-text"
          >
            Date :
          </span>
           
        </p>
      </div>
      <!--v-if-->
      <!--v-if-->
    </div>
  </div>
  <div
    class="row mb-3"
  >
    <app-title-stub />
  </div>
  <div
    class="row artists-list"
  >
    
    
  </div>
</div>
`;
